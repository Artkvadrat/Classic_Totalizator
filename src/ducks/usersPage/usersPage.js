import HTTPService from '../../services/HTTPService';

export const REQUESTED_USERS = 'users_page/requested';
export const RECEIVED_USERS = 'users_page/received';

const requestedUsers = () => ({
  type: REQUESTED_USERS
});

const receivedUsers = (data) => ({
  type: RECEIVED_USERS,
  payload: data
});

export const loadUsers = () => (dispatch) => {
  dispatch(requestedUsers());

  return HTTPService.request({
    path: '/api/v1/account'
  }).then((data) => dispatch(receivedUsers(data)));
};

const initialState = {
  isLoading: false,
  userList: []
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case REQUESTED_USERS:
      return {
        ...state,
        isLoading: true
      };
    case RECEIVED_USERS:
      return {
        ...state,
        isLoading: false,
        userList: action.payload
      };
    default:
      return state;
  }
};

export default reducer;
