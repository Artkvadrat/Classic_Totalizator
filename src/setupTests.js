import enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-17-updated';

enzyme.configure({ adapter: new Adapter() });

const localStorageMock = {
  getItem: jest.fn(),
  setItem: jest.fn(),
  removeItem: jest.fn(),
  clear: jest.fn()
};
window.localStorage = localStorageMock;
