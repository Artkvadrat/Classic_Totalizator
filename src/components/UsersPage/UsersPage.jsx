import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { loadUsers } from '../../ducks/usersPage/usersPage';

const UserPage = () => {
  const dispatch = useDispatch();

  const { isLoading, userList } = useSelector((state) => state.userPage);

  useEffect(() => {
    dispatch(loadUsers());
  }, [dispatch]);
  return <h1>Hello World!</h1>;
};

export default UserPage;
