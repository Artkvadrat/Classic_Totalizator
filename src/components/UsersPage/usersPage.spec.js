import React from 'react';
import { shallow, mount } from 'enzyme';
import { Col, Row, Skeleton, Table } from 'antd';
import { useDispatch, useSelector } from 'react-redux';
import { loadUsers } from '../../ducks/usersPage/usersPage';

import '../../setupTests';

import UsersPage from './UsersPage';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
  useDispatch: jest.fn()
}));

jest.mock('../../ducks/usersPage/usersPage', () => ({
  loadUsers: jest.fn()
}));

describe('UsersPage component', () => {
  const dispatch = jest.fn();
  const mockedLoadUsers = jest.fn();

  beforeAll(() => {
    window.matchMedia =
      window.matchMedia ||
      function () {
        return {
          matches: false,
          addListener: function () {},
          removeListener: function () {}
        };
      };
  });

  beforeEach(() => {
    useDispatch.mockReturnValue(dispatch);
    loadUsers.mockReturnValue(mockedLoadUsers);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render list without errors', () => {
    useSelector.mockReturnValueOnce({ isLoading: false, userList: [] });
    const wrapper = shallow(<UsersPage />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Skeleton without errors', () => {
    useSelector.mockReturnValueOnce({ isLoading: true, userList: [] });
    const wrapper = shallow(<UsersPage />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should dispatch event loadUsers on mount', () => {
    useSelector.mockReturnValueOnce({ isLoading: false, userList: [] });
    const wrapper = mount(<UsersPage />);
    expect(dispatch).toHaveBeenCalled();
    expect(dispatch).toHaveBeenCalledWith(loadUsers());
    expect(loadUsers).toHaveBeenCalled();
  });
});
