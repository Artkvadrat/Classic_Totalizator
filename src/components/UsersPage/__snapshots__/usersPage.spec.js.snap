// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersPage component should render Skeleton without errors 1`] = `
<Skeleton
  active={true}
  avatar={false}
  paragraph={true}
  title={true}
/>
`;

exports[`UsersPage component should render list without errors 1`] = `
<Fragment>
  <Row
    justify="start"
    style={
      Object {
        "margin": 16,
      }
    }
  >
    <Col
      span={20}
    >
      <h1>
        Users
      </h1>
    </Col>
  </Row>
  <Table
    rowKey="key"
  >
    <Column
      dataIndex="dob"
      key="dob"
      title="Creation time"
    />
    <Column
      dataIndex="email"
      key="email"
      title="User Email"
    />
    <Column
      dataIndex="walletAmount"
      key="walletAmount"
      title="Wallet"
    />
  </Table>
</Fragment>
`;
